### ISSUE TITLE
Brave iOS 1.33 "Block Scripts" doesn't block js on first page load

### ISSUE THREAD

**Issue by sinner-** [brave/brave-ios#4819]
### Description: 
I have Brave configured to always be in Private Browsing Only mode and with all shields enabled, including "Block Scripts".

I have a Twitter URL bookmarked (not logged in due to Private Browsing always being enabled), https://mobile.twitter.com/thegrugq - previously I would have to tap the shield and disable "Block Scripts".

As of the most recent update, when I tap the bookmark for thegrugq Twitter page, it loads without JS being blocked! Then if I tap refresh the JS is blocked as expected.

### Steps to Reproduce 
  1. Private Browsing Only mode enabled
  2. Block Scripts enabled
  3. https://mobile.twitter.com/thegrugq bookmarked
  4. Go to the bookmark
  5. Site loads normally, unexpected
  6. Reload, JS is blocked as expected

**Actual result:**
Site loads normally although scripts should be blocked.

**Expected result:**
Site should not load (as it is JS based) until Block Scripts is disabled.

**Reproduces how often:**
Easily Reproduced

**Brave Version:**
 1.33 (21.12.14.16)

**Device details:**
 iPhone SE 1 iOS 15.2

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information
Doesn't seem to affect other bookmarks, e.g. I have https://www.bloomberg.com/quote/XAUAUD:CUR bookmarked and JS is blocked as expected there.

**Comment by diracdeltas** [comment]
i opened a similar report here: https://github.com/brave/brave-ios/issues/3010

the person who was working on it is no longer at brave so i'll try to get this re-prioritized

**Comment by sinner-** [comment]
Thanks Yan, I am a Twitter fan of yours.

**Comment by Uni-verse** [comment]
No repro for me on iPhone 12, iOS 15.1.1 using 1.33 and marketplace app.

@iccub

**Comment by iccub** [comment]
Thanks for checking @Uni-verse,
@sinner- are you able to reproduce it consistently? Is it only for twitter or other websites?

We tried to reproduce that within the iOS team and no luck.

On technical side we use Apple's webkit flag to fully disable javascript 
https://developer.apple.com/documentation/webkit/wkpreferences/1536203-javascriptenabled

**Comment by sinner-** [comment]
Thanks for checking.

Yep I am able to reproduce it every time I try (every day).

Other sites don't seem affected, per original report another URL I have bookmarked is https://www.bloomberg.com/quote/XAUAUD:CUR and JS doesn't load there.

I notice the report from Yan in #3010 is from Nov 2020 but this definitely did not start affecting me back then, I only hit this after updating to 1.33 (21.12.14.16). Upgrading to the newer version that's been released since then hasn't fixed it either.

Is there anything I can do to help provide more evidence/info?

**Comment by iccub** [comment]
@sinner- few things that could help us:
- test it on our public beta build, you can join here https://testflight.apple.com/join/Trcbh1o3
- test on another iOS device if you have any spare ones

- does it happen when you go to twitter manually by typing on url bar or do you have to access it by a bookmark or a new tab page favorite tile?
- does it happen if you are not in private browsing only mode?

**Comment by sinner-** [comment]
Just tested the 3rd one (manual typing) and confirmed it still happens, tried a different twitter URL e.g. https://mobile.twitter.com/jeffsnider_aip

Tomorrow is the weekend here, will try and do some more testing of the other 3 tomorrow.

Thanks again.

**Comment by sinner-** [comment]
Ok sorry for the delay but tested all of this:

1. Still affected on beta build
2. Tested on my partners device, enabled "block scripts" and it worked as expected (JS blocked on first load).
3. As mentioned above it does happen when typing manually
4. I turned off private browsing only mode and it still happens from both a private tab and normal tab. 

Just to clarify again, in all cases, as soon as I reload the site, the JS is blocked as expected. It's the "first load" that is the problem.

So it kind of sounds like something gets wedged on my personal Brave? 

I just completely uninstalled Brave then re-installed from my phone and still get the same issue - how is that possible? 

What else can I do to demonstrate the issue, doesn't look like there is any kind of dump or debug I can take?
